modules/simple_bucket/main.tf

```diff
@@ -114,7 +114,7 @@ resource "google_storage_bucket" "bucket" {
 
 resource "google_storage_bucket_iam_member" "members" {
   for_each = {
-    for m in var.iam_members : "${m.role} ${m.member}" => m
+    for index, m in var.iam_members : "${index} ${m.role}" => m
```

Comment:
The key is not used anywhere. It only needs to be unique.

Reply:
Agreed, but will this change trigger recreating the resource for existing users?

Reply:
+1 to possible recreate issue (although not a huge concern IMO because it's IAM and not resource) and def we should not release as chore:. My other concern is this exposes us to order of variables so any position change is list would could delete and recreate even set is same.

For CR usecase, couldn't we make use of `var.service_account_project_roles` to assign this at project level?

Reply:
@bharathkkb We can use `service_account_project_roles` to provide cloud-run service account permission for GCS. I wanted to utilize the bucket level iam mechanism since it was already available. With pointed regression, have removed this specific change.